lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,6 +56,68 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHead = (doc) ->
+  loadedAssets = extractTrackAssets document
+  fetchedAssets = extractTrackAssets doc
+
+  updateLinkTags(loadedAssets, fetchedAssets)
+  updateScriptTags(loadedAssets, fetchedAssets)
+
+extractTrackAssets = (doc) ->
+  for node in doc.querySelector('head').childNodes when node.getAttribute?('data-turbolinks-track')?
+    node
+
+filterForNodeType = (nodeList, nodeType) ->
+  (node for node in nodeList when node.nodeName == nodeType)
+
+insertScript = (scriptNode) ->
+  newNode = document.createElement 'script'
+  newNode.setAttribute attr.name, attr.value for attr in scriptNode.attributes
+  newNode.appendChild document.createTextNode scriptNode.innerHTML
+
+  deferred = $.Deferred()
```

Comment:
This is the same deal as `.map` and friends, just used things I knew were cross browser because I wasn't sure what turbograft's targets were.

I'm totally down to use `Promise` if we are cool with: 
![image](https://cloud.githubusercontent.com/assets/4889856/17603548/ab1903a2-5fdf-11e6-99bc-3540ab0eadc9.png)

I know we could use bluebird or whatever, but jquery was the only promise shim I noticed already present inside the project and I didn't want to add KB.